harness/tests/cli/util.py

```diff
@@ -151,3 +154,18 @@ def expect_get_info(
         rsps.get(f"{master_url}/info", status=200, json={"version": det.__version__})
     else:
         responses.get(f"{master_url}/info", status=200, json={"version": det.__version__})
+
+
+def check_cli_output(args: List[str], expected: str) -> None:
+    """
+    Helper method to test CLI methods that checks redirected STDOUT from the executed command
+    matches expected output.
+    """
+    with contextlib.redirect_stdout(io.StringIO()) as f:
+        cli.main(args=args)
+    actual = f.getvalue()
+    exp_lines = expected.splitlines(keepends=True)
+    act_lines = actual.splitlines(keepends=True)
+    diff_lines = difflib.ndiff(act_lines, exp_lines)
+    diff = "".join(diff_lines)
+    assert expected == actual, f"CLI output for {args} does not match expected output:\n {diff}"
```

Comment:
You dropped the indicators from my suggestions to help parse what was actual and what was received.  I had:

```python
raise ValueError(f"cli output for {args} output actual (-) != expected (+):\n{diff}")
```

Because that tells you if you see a `-` line, that means it was in actual but not expected, and if you see a `+` line it was in expected but not actual.